src/odemis/gui/cont/multi_point_correlation.py

```diff
@@ -60,6 +60,43 @@ class GridColumns(Enum):
 # Regex search pattern to distinguish between FIB and FM target. These targets can
 # have the same type of Fiducials but there is a prefix in the name to distinguish them.
 FIDUCIAL_PATTERN = r"^[^-]+-"
+REFRACTIVE_SCALE = 0.495  # Refractive index scale factor of the medium (water) for Z correction
+
+
+def update_feature_correlation_target(correlation_target: FIBFMCorrelationData,
+                                      tab_data: CryoGUIData) -> FIBFMCorrelationData:
+    """
+    Update the correlation target with the latest fiducials and POIs from the tab data model.
+    :param correlation_target: (FIBFMCorrelationData) the correlation target to update
+    :param tab_data: (CryoGUIData) the tab data model
+    :return: (FIBFMCorrelationData) the updated correlation target
+    """
+    # Corner case: When fiducials are deleted and the indices are not continuous. Then the fiducial pairs
+    # will be incorrectly matched together.
+    # TODO Handle the corner case
+    fib_fiducials = []
+    fm_fiducials = []
+    correlation_target.fm_pois = []
+    for target in tab_data.main.targets.value:
+        if target.type.value == TargetType.FibFiducial:
+            fib_fiducials.append(target)
+        elif target.type.value == TargetType.Fiducial:
+            fm_fiducials.append(target)
+        elif target.type.value == TargetType.PointOfInterest:
+            correlation_target.fm_pois.append(target)
+        elif target.type.value == TargetType.SurfaceFiducial:
+            correlation_target.fib_surface_fiducial = target
```

Comment:
I see a some `if, elif statements for targets` in the code, maybe you need a `TargetManager` to handle the logic inside it